ssz_serialization/merkleization.nim

```diff
@@ -583,53 +592,35 @@ func hashTreeRootAux[T](x: T): Digest =
           pos += sizeof(E)
     else:
       trs "FIXED TYPE; USE CHUNK STREAM"
-      var merkleizer = createMerkleizer(maxChunksCount(T, Limit x.len))
-      chunkedHashTreeRootForBasicTypes(merkleizer, x)
-  elif T is BitArray:
-    hashTreeRootAux(x.bytes)
+      const totalChunks = maxChunksCount(T, x.len)
+      chunkedHashTreeRoot(totalChunks, x)
+  elif T is List:
+    const totalChunks = maxChunksCount(T, x.maxLen)
+    chunkedHashTreeRoot(totalChunks, asSeq x)
+      .mixInLength(x.len)
   elif T is SingleMemberUnion:
     doAssert x.selector == 0'u8
     merkleizeFields(Limit 2):
       addField hashTreeRoot(toSszType(x.value))
-  elif T is array|object|tuple:
+  elif T is object|tuple:
```

Comment:
The array of not element with `BasicType` now takes a different path of which I can't immediately tell whether it does the same.

Reply:
Yep, the details can be seen isolated in the commit `use chunkedHashTreeRoot for both basic and non-basic`. All arrays go through `chunkedHashTreeRoot` now.

Reply:
I'll optimize this commit for code-size as well, and submit as separate PR.